packages/web3-eth/src/utils/format_transaction.ts

```diff
@@ -40,7 +40,11 @@ export function formatTransaction<
 
 	formattedTransaction = format(options.transactionSchema, formattedTransaction, returnFormat);
 
-	if (!isNullish(formattedTransaction.data) && !isNullish(formattedTransaction.input))
+	if (
+		!isNullish(formattedTransaction.data) &&
+		!isNullish(formattedTransaction.input) &&
+		formattedTransaction.data !== formattedTransaction.input
```

Comment:
codecov patch diff is not passing, due to following not covered.

Reply:
The new line is covered by the e2e tests, but I can add a unit test